packages/server/src/automations/tests/scenarios/scenarios.spec.ts

```diff
@@ -233,4 +319,168 @@ describe("Automation Scenarios", () => {
+            type: FieldType.NUMBER,
+            constraints: {
+              presence: true,
+            },
+          },
+        },
+      })
+    })
+
+    it("should stop an automation if the condition is not met", async () => {
+      const builder = createAutomationBuilder({
+        name: "Test Equal",
+      })
+
+      const results = await builder
+        .appAction({ fields: {} })
+        .createRow({
+          row: {
+            name: "Equal Test",
+            value: 10,
+            tableId: table._id,
+          },
+        })
+        .queryRows({
+          tableId: table._id!,
+        })
+        .filter({
+          field: "{{ steps.2.rows.0.value }}",
+          condition: FilterConditions.EQUAL,
+          value: 20,
+        })
+        .serverLog({ text: "Equal condition met" })
+        .run()
+
+      expect(results.steps[2].outputs.success).toBeTrue()
+      expect(results.steps[2].outputs.result).toBeFalse()
+      expect(results.steps[3]).toBeUndefined()
+    })
+
+    it("should continue the automation if the condition is met", async () => {
+      const builder = createAutomationBuilder({
+        name: "Test Not Equal",
+      })
+
+      const results = await builder
+        .appAction({ fields: {} })
+        .createRow({
+          row: {
+            name: "Not Equal Test",
+            value: 10,
+            tableId: table._id,
+          },
+        })
+        .queryRows({
+          tableId: table._id!,
+        })
+        .filter({
+          field: "{{ steps.2.rows.0.value }}",
+          condition: FilterConditions.NOT_EQUAL,
+          value: 20,
+        })
+        .serverLog({ text: "Not Equal condition met" })
+        .run()
+
+      expect(results.steps[2].outputs.success).toBeTrue()
+      expect(results.steps[2].outputs.result).toBeTrue()
+      expect(results.steps[3].outputs.success).toBeTrue()
+    })
+
+    const testCases = [
+      {
+        condition: FilterConditions.EQUAL,
+        value: 10,
+        rowValue: 10,
+        expectPass: true,
+      },
+      {
+        condition: FilterConditions.NOT_EQUAL,
+        value: 10,
+        rowValue: 20,
+        expectPass: true,
+      },
+      {
+        condition: FilterConditions.GREATER_THAN,
+        value: 10,
+        rowValue: 15,
+        expectPass: true,
+      },
+      {
+        condition: FilterConditions.LESS_THAN,
+        value: 10,
+        rowValue: 5,
+        expectPass: true,
+      },
+      {
+        condition: FilterConditions.GREATER_THAN,
+        value: 10,
+        rowValue: 5,
+        expectPass: false,
+      },
+      {
+        condition: FilterConditions.LESS_THAN,
+        value: 10,
+        rowValue: 15,
+        expectPass: false,
+      },
+    ]
+
+    testCases.forEach(({ condition, value, rowValue, expectPass }) => {
+      it(`should ${
+        expectPass ? "pass" : "fail"
+      } the filter when condition is "${condition}" and value is ${value}`, async () => {
```

Comment:
nit: prefer `it.each(testCases)("...")`, if the reason you didn't do this is the ternary inside of the string I wouldn't worry too much about grammatically correct test names.